operator/pkg/ciliumidentity/controller.go

```diff
@@ -114,14 +112,13 @@ func registerController(p params) {
 	p.Lifecycle.Append(cidController)
 }
 
-func (c *Controller) Start(_ cell.HookContext) error {
+func (c *Controller) Start(ctx cell.HookContext) error {
 	c.logger.Info("Starting CID controller Operator")
 	defer utilruntime.HandleCrash()
 
 	var err error
-	c.context, c.contextCancel = context.WithCancel(context.Background())
 	c.reconciler, err = newReconciler(
-		c.context,
+		ctx,
```

Comment:
This `ctx` is saved into the reconciler object and used after this Start hook completes in both `createCID` and `updateCID`. Unfortunately this is not going to work, since that context is cancelled after all the hive start hooks complete:

https://github.com/cilium/cilium/blob/b54391022a859335d94838f479d595b76f897e60/vendor/github.com/cilium/hive/cell/lifecycle.go#L89-L93

We should use the context from the job managing `runResourceWorker`

Reply:
Indeed, you're right, missed somehow that part, https://github.com/cilium/hive/blob/main/cell/lifecycle.go#L93

Reply:
Hey @pippolo84, can you have a second look over? I've mode the initReconciler logic inside the work group job group. I could've extracted only the .Store() relevant logic but I don't see much different as the  `newReconciler` is not doing much work that could be blocking and timeout. Let me know what you think.

Reply:
I think it's fine since it is safe to add jobs after the group has already started. :+1: